modules/data_release/jsx/uploadFileForm.js

```diff
@@ -0,0 +1,196 @@
+          </strong>
+        </div>
+      );
+    }
+
+    return (
+      <FormElement
+        name='uploadfile'
+        fileUpload={true}
+        onSubmit={this.validateAndSubmit}>
+        <FileElement
+          name='file'
+          label='File to upload'
+          onUserInput={this.updateFormElement}
+          required={true}
+          value={this.state.formData.file}
+        />
+        <TextboxElement
+          name='version'
+          label='Version'
+          onUserInput={this.updateFormElement}
+          required={false}
+          value={this.state.formData.version}
+        />
+        <ButtonElement label='Upload File'/>
+        <div className='row'>
+          <div className='col-sm-9 col-sm-offset-3'>
+            <ProgressBar value={this.state.uploadProgress}/>
+          </div>
+        </div>
+      </FormElement>
+    );
+  }
+
+  /**
+   * Update a form Element
+   *
+   * @param {string} formElement - name of the selected element
+   * @param {string} value - selected value for corresponding element
+   */
+  updateFormElement(formElement, value) {
+    let formData = this.state.formData;
+
+    if (value === '') {
+      formData[formElement] = null;
+    } else {
+      formData[formElement] = value;
+    }
+
+    this.setState({formData: formData});
+  }
+
+  /**
+   * Validate and submit the upload
+   */
+  validateAndSubmit() {
+    let files = this.state.data.files ? this.state.data.files : [];
+    let formData = this.state.formData;
+
+    let errorMessage = {
+      Filename: undefined,
+    };
+
+    let hasError = {
+      Filename: undefined,
+    };
+
+    if (!formData.file) {
+      errorMessage.Filename = 'You must select a file to upload';
+      hasError.Filename = true;
+      this.setState({errorMessage, hasError});
+      return;
+    }
+
+    // Grep the uploaded file name
+    let fileName = formData.file ? formData.file.name.replace(/\s+/g, '_') : null;
+
+    // Check for duplicate file names
+    let isDuplicate = files.indexOf(fileName);
+    if (isDuplicate >= 0) {
+      swal({
+        title: 'Are you sure?',
+        text: 'A file with this name already exists!\n Would you like to override existing file?',
+        type: 'warning',
+        showCancelButton: true,
+        confirmButtonText: 'Yes, I am sure!',
+        cancelButtonText: 'No, cancel it!',
+      }, function(isConfirm) {
+        if (isConfirm) {
+          this.uploadFile();
+        } else {
+          swal('Cancelled', 'Your imaginary file is safe :)', 'error');
+        }
+      }.bind(this));
+    } else {
+      this.uploadFile();
+    }
+  }
+
+  /**
+   * Upload the file to the server
+   */
+  uploadFile() {
+    let formData = this.state.formData;
+    let formObj = new FormData();
+    for (let key in formData) {
+      if (formData[key] !== '') {
+        formObj.append(key, formData[key]);
+      }
+    }
+
+    // fetch API to upload the file
+    fetch(this.props.action, {
```

Comment:
@HenriRabalais @driusan I still have an issue with that fetch for the file upload.

The file name is correctly passed to the PHP side, however, the rest of the $_FILES['file'] object sent to the PHP side is filled with '\n'... Including the 'tmp_name' key which is then used by the PHP side to move the file to the user_uploads directory of the module.

Any idea what I am doing wrong? I googled a lot but could not find the solution on my own. Your help with this would be greatly appreciated :). Thanks!

Reply:
PS: all the other ajax requests have been replaced by `fetch` BTW. This is the only one giving me trouble...

Reply:
@cmadjar perhaps i can come take a look at it offline next week

Reply:
@HenriRabalais As you probably guessed by now, I was away ;). If you are back this week, maybe we could take a look at it together tomorrow (Tuesday)?

Thank you and happy new year!!